R/data-storage-log-file.R

```diff
@@ -99,24 +108,37 @@ DataStorageLogFile <- R6::R6Class( # nolint object_name_linter
     read_data = function(date_from, date_to, bucket) {
 
       checkmate::assert_string(bucket)
-      checkmate::assert_date(date_from)
-      checkmate::assert_date(date_to)
+      checkmate::assert_date(date_from, null.ok = TRUE)
+      checkmate::assert_date(date_to, null.ok = TRUE)
 
       if (!file.exists(bucket)) {
         return(
           private$table_schema
         )
       }
 
-      readLines(bucket) %>%
+      result <- readLines(bucket) %>%
         lapply(function(x) jsonlite::fromJSON(x, flatten = TRUE)) %>%
         dplyr::bind_rows() %>%
-        dplyr::filter(
-          time >= date_from,
-          time <= date_to
-        ) %>%
+        dplyr::bind_rows(private$table_schema) %>%
+        dplyr::mutate(time = lubridate::as_datetime(.data$time, tz = "UTC"))
+
+      if (!is.null(date_from)) {
+        result <- dplyr::filter(
+          result,
+          .data$time >= lubridate::as_datetime(date_from, tz = "UTC")
+        )
+      }
+      if (!is.null(date_to)) {
+        result <- dplyr::filter(
+          result,
+          .data$time <= lubridate::as_datetime(date_to, tz = "UTC")
```

Comment:
Wow!! Thanks a million for catching this.

I suggest using dates to compare instead of datetime, I.e. _(check image below)_

![image](https://user-images.githubusercontent.com/211358/231992134-d2b9bbde-bbf9-4acd-92ff-46424ae1e29a.png)

In LogFile that means:

```
lubridate::as_date(.data$time) >= lubridate::as_date(date_from)
```

in SQL it means we compare `time (seconds since epoch) < (date_to + 1 %>% seconds())` which will take all records before midnight of the `date_to` date.